client/editor/editor.go

```diff
@@ -155,41 +181,58 @@ func (e *Editor) Draw() {
 		}
 	}
 
-	if e.ShowMsg {
-		e.SetStatusBar()
-	} else {
-		e.showPositions()
-	}
+	e.DrawStatusBar()
 
 	// Flush back buffer!
 	termbox.Flush()
 }
 
-// SetStatusBar sets the message (e.StatusMsg) in the editor's status bar.
-// The message disappears automatically after 5 seconds, in order to simulate the "popup" effect.
-func (e *Editor) SetStatusBar() {
-	e.ShowMsg = true
+// DrawStatusBar shows all status and debug information on the bottom line of the editor.
+func (e *Editor) DrawStatusBar() {
+	if e.ShowMsg {
+		e.DrawStatusMsg()
+	} else {
+		e.DrawInfoBar()
+	}
+
+	// Render connection indicator
+	if e.IsConnected {
+		termbox.SetBg(e.Width-1, e.Height-1, termbox.ColorGreen)
+	} else {
+		termbox.SetBg(e.Width-1, e.Height-1, termbox.ColorRed)
+	}
+}
 
+// DrawStatusMsg draws the editor's status message at the bottom of the
+// termbox window.
+func (e *Editor) DrawStatusMsg() {
 	for i, r := range []rune(e.StatusMsg) {
 		termbox.SetCell(i, e.Height-1, r, termbox.ColorDefault, termbox.ColorDefault)
 	}
-
-	// Toggle showMsg to false to hide the message.
-	_ = time.AfterFunc(5*time.Second, func() {
-		e.ShowMsg = false
-	})
 }
 
-// showPositions shows the cursor positions with other details.
-func (e *Editor) showPositions() {
+// DrawInfoBar draws the editor's debug information and the names of the
+// active users in the editing session at the bottom of the termbox window.
+func (e *Editor) DrawInfoBar() {
 	x, y := e.calcCursorXY(e.Cursor)
 
-	// Construct message for debugging.
-	str := fmt.Sprintf("x=%d, y=%d, cursor=%d, len(text)=%d", x, y, e.Cursor, len(e.Text))
+	i := 0
+	for j, user := range e.Users {
+		for _, r := range user {
+			termbox.SetCell(i, e.Height-1, r, userColors[j], termbox.ColorDefault)
```

Comment:
What happens when the number of users exceed the number of pre-defined user colors? I believe a modulus operator could help here (to loop back to zero)?

Reply:
Ah of course, thank you!